test-app-reactive/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagenameasync/integration/TelemetryPropagationTest.kt

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.digital.hmpps.hmppstemplatepackagenameasync.integration
+
+import io.opentelemetry.api.GlobalOpenTelemetry
+import io.opentelemetry.api.OpenTelemetry
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.api.trace.SpanContext
+import io.opentelemetry.api.trace.TraceFlags
+import io.opentelemetry.api.trace.TraceState
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.propagation.ContextPropagators
+import kotlinx.coroutines.test.runTest
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.kotlin.await
+import org.awaitility.kotlin.matches
+import org.awaitility.kotlin.untilCallTo
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue
+import software.amazon.awssdk.services.sns.model.PublishRequest
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest
+import uk.gov.justice.digital.hmpps.hmppstemplatepackagenameasync.service.HmppsEvent
+import uk.gov.justice.digital.hmpps.hmppstemplatepackagenameasync.service.Message
+import uk.gov.justice.hmpps.sqs.countMessagesOnQueue
+
+class TelemetryPropagationTest : IntegrationTestBase() {
+  @BeforeEach
+  fun setup() {
+    GlobalOpenTelemetry.resetForTest()
+    GlobalOpenTelemetry.set(OpenTelemetry.propagating(ContextPropagators.create(W3CTraceContextPropagator.getInstance())))
+  }
+
+  @Test
+  fun `telemetry information is propagated between publishers and listeners`() = runTest {
```

Comment:
Is there any way we could test the PUBLISH and RECEIVE spans are created, for regression?

Reply:
Good idea - I've added a check for the exported spans at the end of the test.